examples/cryptocurrency/src/lib.rs

```diff
@@ -210,20 +251,29 @@ impl Transaction for TxTransfer {
     /// [`TxCreateWallet`]: struct.TxCreateWallet.html
     fn execute(&self, view: &mut Fork) -> ExecutionResult {
         let mut schema = CurrencySchema::new(view);
-        let sender = schema.wallet(self.from());
-        let receiver = schema.wallet(self.to());
-        if let (Some(sender), Some(receiver)) = (sender, receiver) {
-            let amount = self.amount();
-            if sender.balance() >= amount {
-                let sender = sender.decrease(amount);
-                let receiver = receiver.increase(amount);
-                println!("Transfer between wallets: {:?} => {:?}", sender, receiver);
-                let mut wallets = schema.wallets_mut();
-                wallets.put(self.from(), sender);
-                wallets.put(self.to(), receiver);
-            }
+
+        let sender = match schema.wallet(self.from()) {
+            Some(val) => val,
+            None => Err(Error::SenderNotExist)?,
+        };
+
+        let receiver = match schema.wallet(self.to()) {
+            Some(val) => val,
+            None => Err(Error::ReceiverNotExist)?,
+        };
+
+        let amount = self.amount();
+        if sender.balance() >= amount {
+            let sender = sender.decrease(amount);
+            let receiver = receiver.increase(amount);
```

Comment:
If I understand correctly, this instruction may lead to a panic (`u64` overflow, to be exact). I'm not quite sure whether this should be an error, or panicking is OK in this case.

Reply:
As for me, panic is OK here.